examples/encoding/16_dash_on_demand.js

```diff
@@ -0,0 +1,515 @@
+  };
+
+  return bitmovin.encoding.manifests
+    .dash(manifest.id)
+    .periods(period.id)
+    .adaptationSets(adaptationSet.id)
+    .representations.mp4.add(representation);
+};
+
+const createManifest = (output, videoManifestRepresentations, audioManifestRepresentations) =>
+  new Promise(resolve => {
+    return createDashManifest(output).then(createdManifest => {
+      return addPeriodToManifest(createdManifest).then(createdPeriod => {
+        const videoRepresentationPromises = addVideoAdaptationSetToPeriod(createdManifest, createdPeriod).then(
+          createdVideoAdaptationSet => {
+            return Promise.map(
+              videoManifestRepresentations,
+              videoRepresentation => {
+                return addDashMp4RepresentationToAdaptationSet(
+                  createdManifest,
+                  createdPeriod,
+                  createdVideoAdaptationSet,
+                  videoRepresentation.encoding,
+                  videoRepresentation.muxing,
+                  videoRepresentation.profile
+                );
+              },
+              {concurrency: 1}
+            );
+          }
+        );
+        const audioRepresentationPromises = addAudioAdaptationSetToPeriod(createdManifest, createdPeriod).then(
+          createdAudioAdaptationSet => {
+            return Promise.map(
+              audioManifestRepresentations,
+              audioRepresentation => {
+                return addDashMp4RepresentationToAdaptationSet(
+                  createdManifest,
+                  createdPeriod,
+                  createdAudioAdaptationSet,
+                  audioRepresentation.encoding,
+                  audioRepresentation.muxing,
+                  audioRepresentation.profile
+                );
+              },
+              {concurrency: 1}
+            );
+          }
+        );
+        return Promise.all([videoRepresentationPromises, audioRepresentationPromises]).then(() => {
+          return resolve(createdManifest);
+        });
+      });
+    });
+  });
+
+const waitUntilEncodingFinished = encoding => {
+  return new Promise((resolve, reject) => {
+    const waitForEncodingToBeFinishedOrError = () => {
+      console.log('GET STATUS FOR ENCODING WITH ID ', encoding.id);
+      bitmovin.encoding
+        .encodings(encoding.id)
+        .status()
+        .then(response => {
+          console.log('Encoding status is ' + response.status);
+
+          if (response.status === 'FINISHED') {
+            return resolve(response.status);
+          }
+
+          if (response.status === 'ERROR') {
+            return reject(response.status);
+          }
+
+          setTimeout(waitForEncodingToBeFinishedOrError, 10000);
+        });
+    };
+    waitForEncodingToBeFinishedOrError();
+  });
+};
+
+const startEncodingAndWaitForItToBeFinished = encoding => {
+  const startPromise = bitmovin.encoding.encodings(encoding.id).start();
+
+  return new Promise((resolve, reject) => {
```

Comment:
Please try to avoid `new Promise` here you could do `return startPromise....` and in the `.then()` `return waitUnitlEncodingFinished...`